code/DynamicCache.php

```diff
@@ -384,6 +393,10 @@ public function run($url)
         // This prevents a new user's security token from being regenerated incorrectly
         $_SESSION['SecurityID'] = SecurityToken::getSecurityID();
 
+        // Set the stage of the website
+        // This is normally called in VersionedRequestFilter.
+        Versioned::choose_site_stage();
```

Comment:
@tractorcow Can confirm, ?stage=Stage will kick a user to the login page even with a cached live copy.

If we decided to cache Stage / Live seperately however (ie. enabled caching for ?stage=Stage), then an unauthed user could see the content.